internal/components/component.go

```diff
@@ -113,3 +116,16 @@ func ConstructServicePort(current *corev1.ServicePort, port int32) corev1.Servic
 		Protocol:    current.Protocol,
 	}
 }
+
+func GetPortsForConfig(logger logr.Logger, config map[string]interface{}, retriever ParserRetriever) ([]corev1.ServicePort, error) {
+	var ports []corev1.ServicePort
+	for componentName, componentDef := range config {
```

Comment:
This is really the type of this component, not the name.

Reply:
This is the name because it's what we're getting from the config which could be the form `name=type/nameoverride`

Reply:
In that case, the collector term for this is the `component ID`, and we should use it as well. Doesn't necessarily have to be this PR.